### DISCUSSION TITLE
Pass data from php to scripts loaded by block.json

### DISCUSSION THREAD

**Discussion by swpw** [WordPress/gutenberg#47074] [Q&A]
Hey,
During block render I generate random block id. How would i go around passing the id to scripts loaded by block.json. 

In case of having the same multiple blocks on the page I want to enqueue script files each time the block is added to the page therefore each script within that block can read unique id of that block.

As of now i managed to pass id to block scripts using this:
```php
add_filter('script_loader_tag', function ($tag, $handle, $src) use ($block_data) {

	// $block_data['script_handles'] = block.json list of scripts to load
	foreach ($block_data['script_handles'] as $script) {
			if ($script === $handle) {
					$tag = '<script type="module">const block_id = "' . $block_data['id'] . '";' . file_get_contents($src) . '</script>';
			}
	}

	return $tag;
}, 10, 3);
```
But as far as it does pass the id to the scripts, It only works for the first block. Every next block doesn't load the script since it never gets enqueued as it's a duplicate. How could i go around this problem?

**Comment by t-hamano** [comment]
Hi @swpw,
What is the purpose of loading a unique ID? Depending on the purpose, there might be a more reasonable approach, such as a JavaScript filter hook.
